post-processor/docker-push/post-processor.go

```diff
@@ -105,5 +106,10 @@ func (p *PostProcessor) PostProcess(ui packer.Ui, artifact packer.Artifact) (pac
 		return nil, false, err
 	}
 
+	if p.config.Remove {
+		ui.Say(fmt.Sprintf("Docker image: %s will be removed", name))
+		return artifact, false, nil
```

Comment:
This looks weird, why does this propagate the source artifact but not the other branch?

Reply:
I'm sorry I can't understand what mean "other branch". I guess that is about the keep artifact value, I return false because in this way the destroy method will be applied by the [priorArtifact](https://github.com/hashicorp/packer/blob/master/builder/docker/artifact_import.go#L36). The propagate artifact is only to [skip](https://github.com/hashicorp/packer/blob/master/packer/build.go#L259-L262) the halting post-processor chain.

Reply:
other branch as in code. If remove is false, why doesn't it return the artifact

Reply:
Oh, I got it. I didn't propagate the artifact because I want to maintain the actual flow that is the breaking the loop, but you are correct that this look weird and an option could be to return the artifact and return keep as true, when the remove is false, this will look like this:

```
keep:= !p.config.Remove
if p.config.Remove {
   ui.Say(fmt.Sprintf("Docker image: %s will be removed", name))
}
return artifact, keep, nil
```
What do you think about this?